packages/calendar/src/Calendar.ts

```diff
@@ -169,16 +161,80 @@ export class Calendar extends SpectrumElement {
         document.removeEventListener('mousedown', this.resetDateFocusIntent);
     }
 
-    override willUpdate(changedProperties: PropertyValues): void {
-        if (changedProperties.has('min')) this.setMinCalendarDate();
+    public clear(): void {
+        this.value = undefined;
+    }
+
+    constructor() {
+        super();
+        this.setNumberFormatter();
+        this.setWeekdays();
+        this.setCurrentMonthDates();
+    }
+
+    private convertToCalendarDates(): void {
+        this.min = this.min && toCalendarDate(this.min);
+        this.max = this.max && toCalendarDate(this.max);
+        this.value = this.value && toCalendarDate(this.value);
+    }
+
+    private checkDatesCompliance(
+        changesValue: boolean,
+        changesMin: boolean,
+        changesMax: boolean
+    ): void {
+        this.convertToCalendarDates();
+
+        if ((changesMin || changesMax) && this.min && this.max) {
+            const isValidInterval = this.min.compare(this.max) < 0;
+            if (!isValidInterval) {
+                window.__swc.warn(
+                    this,
+                    `<${this.localName}> expects the 'min' to be less than 'max'. Please ensure that 'min' property's date is earlier than 'max' property's date.`,
+                    'https://opensource.adobe.com/spectrum-web-components/components/calendar' // TODO: update link
+                );
+                this.min = undefined;
+                this.max = undefined;
+            }
+        }
 
-        if (changedProperties.has('max')) this.setMaxCalendarDate();
+        if (changesValue && this.value) {
```

Comment:
If only the interval gets changed by the consumer, the component's value should still be checked against said interval. This applies for the DateTimePicker as well and will be modified in an upcoming PR with unit tests for it.
 
```suggestion
if (this.value)
```

Reply:
Fixed in the next PR.